### PR TITLE
Migrate Tagit initialisation to a Stimulus Controller

### LINKED ISSUE #10100: 🎛️ Migrate Tag Widget to a Stimulus Controller `w-tag`
> ℹ️ **Part of the [Stimulus  🎛️ RFC 78](https://github.com/wagtail/rfcs/pull/78)**

### Is your proposal related to a problem?

One of the largest blockers to CSP compliance is the overuse of inline script tags, and one of the largest uses of that approach is the taggit field.

It would be good to migrate the taggit field to a Stimulus controller, preserving the jQuery widget approach for now but just ensuring that the initialisation of the field with JS is done via Stimulus.

### Describe the solution you'd like

Create a new Stimulus Controller `TagController` that has the following specification;

* value: `url` (string, required) - the `autocomplete_url` in use today
* value: `allowSpaces` (boolean, not required, default is False
* value: `tagLimit` (number, not required, default - be careful here, we want to default to 0 but us that to put `null` into the init function)
* value: `strict` (boolean, default false) same as `autocompleteOnly` but a shorter name
* Note: We may want to support future options so add a link to the docs in the Controller code https://github.com/aehlke/tag-it/blob/master/README.markdown#options
* We will need to support backwards compatibility - this can be done in the `afterLoad` method, similar to how we are doing this for ProgressController - search for things that look like they could be the taggit field but do not have the data controller attributes.
* The current Tag widget in Python is located in `wagtail/admin/widgets/tags.py`
* The current HTML set up for the general usage of the tag widget is located in `wagtail/admin/templates/wagtailadmin/widgets/tag_widget.html`
* A good start would be to update the Django widget to use a data attributes approach only that will connect to the Stimulus Controller.
* We should be able to remove the `context["widget"]["autocomplete_url"]` & `context["widget"]["options_json"]` and replace them only with plain `attrs` (data attributes).

### Describe alternatives you've considered

* Leave as is, or go all in with finding a new taggit replacement. Hopefully this issue is a step in the right direction.

### Additional context

* Tag fields are used in a lot of places, start small and ensure that we have a solid solution to support existing code approaches used by other packages
* This will be able to get a massive amount of progress towards #7053

### PR DESCRIPTION
**Pull request by lb-**
- Fixes #10100
- Does not migrate away from using a jQuery UI widget
- Preserves the existing global `initTagField` for now (can be removed in a future release)
- [x] Fix unit tests (python)
- [x] Add release considerations
- [x] Move `domReady` to shared util

### CODE REVIEW

**Comment by lb-** [comment]
Ready for review, I plan to pull out the `domReady` promise util out to a shared util but ready for feedback from someone in the core team.

Would love your thoughts @Scotchester  or @thibaudcolas  - hopefully this is not a too controversial one.

As per original issue raised, I opted to not find a jQuery UI tag replacement yet, better we do that as an isolated bit of work. Thankfully this pulls out a large amount of CSP non-compliance and hopefully is a much simpler/secure (escpaing handled 100% at the widget side, no need to worry about it on the template side) way to provide the taggit widget.

File: docs/releases/5.0.md [hunk 1]
```diff
@@ -141,6 +141,40 @@ If there are custom styles in place for the `ModelAdmin`'s header content or mor
...
+
+If the widget is being instantiated in JavaScript or HTML with the global util `window.initTagField`, this undocumented util should be replaced with the new
+`data` attributes only approach.
+
+The global util will be removed in a future release, additionally any direct usage of the jQuery widget in JavaScript (e.g. `$('#my-element).tagit()`) should be removed.
+
+#### Old syntax
+
+```html
+<input id="id_tags" type="text" value="popular, technology" hidden />
```
Comment (nmorduch) [inline docs/releases/5.0.md:156]: Nitpicky small comment: I would find this easier to compare if there were line breaks like in the example for the new syntax

Comment (lb-) [reply docs/releases/5.0.md:156]: Good call. I originally had the others all in one line but after I ran prettier it wrapped the below attributes. 

I'll make the old syntax match this also.

Comment (lb-) [reply docs/releases/5.0.md]: This is done. Thanks @nmorduch

**Comment by lb-** [comment]
Rebased, fixed documentation concern, added an extra recommendation to use the `AdminTagWidget` instead. Additionally moved the `data-controller` declaration to the widgets' init call. This makes more sense as it's a bit more core to the widget behaviour, similar to an id.

**Comment by lb-** [comment]
Argh. Ok I'll try to fix this tomorrow - the testing regex to pull out attributes is to fragile. Moving the data-controller attribute is causing it to fail.

**Comment by lb-** [comment]
Good to go again for review now

**Comment by lb-** [comment]
Rebased with upgrade considerations moved to 5.1 release notes.

**Comment by lb-** [comment]
Rebased & upgrade considerations moved to 5.1 release notes.

File: client/src/utils/domReady.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Returns a promise that resolves once the DOM is ready for interaction.
+ */
+const domReady = () =>
```
[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/domReady.ts:4]: No problems.

[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/domReady.ts:4]: It shouldn't change how the function is used though, but yeah I'll add Async/await to this function.

**Comment by lb-** [comment]
Thanks for the feedback @laymonage  - ready for another review when you have some time.

**Review by laymonage — APPROVED** [review]
Looking great, thanks! Love seeing the inline scripts removal.

**Pull request merged**